ocs_ci/ocs/container.py

```diff
@@ -0,0 +1,315 @@
+        self.sc_data['parameters']['fsName'] = get_cephfs_name()
+        self.sc_data['provisioner'] = defaults.CEPHFS_PROVISIONER
+        self.sc_data['parameters']['pool'] = get_cephfs_data_pool_name()
+        self._setup_sc()
+
+    def _setup_sc(self):
+        """
+        Setup reminder of storage class data
+        """
+
+        if self.sc_name is None:
+            self.sc_name = create_unique_resource_name('test', 'storageclass')
+        self.sc_data['metadata']['name'] = self.sc_name
+        self.sc_data['metadata']['namespace'] = defaults.ROOK_CLUSTER_NAMESPACE
+        self.sc_data['parameters'][
+            'csi.storage.k8s.io/provisioner-secret-name'
+        ] = self.secret_data_name
+        self.sc_data['parameters'][
+            'csi.storage.k8s.io/provisioner-secret-namespace'
+        ] = defaults.ROOK_CLUSTER_NAMESPACE
+
+        self.sc_data['parameters']['clusterID'] = defaults.ROOK_CLUSTER_NAMESPACE
+        self.sc_data['reclaimPolicy'] = self.reclaim_policy
+
+    def _setup_pvc(self):
+        """
+         Setup PVC data
+        """
+        self.pvc_data = templating.load_yaml_to_dict(constants.CSI_PVC_YAML)
+        if self.pvc_name is None:
+            self.pvc_name = create_unique_resource_name('test', 'pvc')
+        self.pvc_data['metadata']['name'] = self.pvc_name
+        self.pvc_data['metadata']['namespace'] = self.namespace
+        self.pvc_data['spec']['accessModes'] = [self.access_mode]
+        self.pvc_data['spec']['storageClassName'] = self.sc_name
+        if self.size:
+            self.pvc_data['spec']['resources']['requests']['storage'] = self.size
+
+    def _create_rbd_pvc(self):
+        """
+         Create a PVC backed by RBD
+        """
+        self.rbd_secret_obj = self._create_ocs(self.secret_data)
+        self.ceph_block_pool_obj = self._create_ocs(self.block_data)
+        self.rbd_sc_obj = self._create_ocs(self.sc_data)
+        if self.num_pvc == 1:
+            self.pvc_obj = self._create_ocs(self.pvc_data, constants.STATUS_BOUND)
+        else:
+            self._create_n_pvc()
+
+    def _create_fs_pvc(self):
+        """
+         Create a PVC backed by CephFS
+        """
+        self.cephfs_secret_obj = self._create_ocs(self.secret_data)
+        self.fs_sc_obj = self._create_ocs(self.sc_data)
+        if self.num_pvc == 1:
+            self.pvc_obj = self._create_ocs(self.pvc_data, constants.STATUS_BOUND)
+        else:
+            self._create_n_pvc()
+
+    def _create_n_pvc(self):
+        with concurrent.futures.ThreadPoolExecutor() as exec:
+            pvc_to_num = {exec.submit(self._create_base_pvc): pvc for pvc in range(self.num_pvc)}
+            for future in concurrent.futures.as_completed(pvc_to_num):
+                try:
+                    future.result()
+                except Exception as exc:
+                    raise exc
+
+    def _create_base_pvc(self):
+        self.pvc_objs = []
+        self.pvc_time_taken = []
+        ts = time.time()
```

Comment:
shouldn't this be before line 225? In any case, as Raz mentioned, parsing logs will be a perfect way.

Reply:
I am going to remove this as its not accurate based on time.